Sources/Workspace/UserToolchain.swift

```diff
@@ -132,35 +131,44 @@ public final class UserToolchain: Toolchain {
     /// Environment to use when looking up tools.
     private let processEnvironment: [String: String]
 
-    public func getClangCompiler() throws -> AbsolutePath {
+    public func getTool(toolName: String) throws -> AbsolutePath {
 
-        if let clangCompiler = _clangCompiler {
-            return clangCompiler
+        if let toolPath = toolPaths[toolName] {
+            return toolPath
         }
 
-        let clangCompiler: AbsolutePath
-
-        // Find the Clang compiler, looking first in the environment.
-        if let value = UserToolchain.lookup(variable: "CC", searchPaths: envSearchPaths) {
-            clangCompiler = value
-        } else {
-            // No value in env, so search for `clang`.
-            let foundPath = try Process.checkNonZeroExit(arguments: whichClangArgs, environment: processEnvironment).chomp()
-            guard !foundPath.isEmpty else {
-                throw InvalidToolchainDiagnostic("could not find `clang`")
+        let toolPath: AbsolutePath
+
+        if toolName == "clang" {
```

Comment:
We shouldn't do conditionalize in middle of this method. I would suggest creating a mapping of tool and its corresponding env variable. That we, it can work for all tools without having such checks.

Reply:
done

Reply:
done